### PR TITLE
Update AWS vendored imports + add elasticloadbalancerv2

### PR DESCRIPTION
**Pull request by andrewvc**
Since upgrading these deps in https://github.com/elastic/beats/pull/12401 is too much for one PR, we're combining them.

### CODE REVIEW

File: vendor/vendor.json [hunk 1]
```diff
@@ -367,6 +385,12 @@
 			"version": "v2.0.0-preview.5",
 			"versionExact": "v2.0.0-preview.5"
 		},
+		{
+			"checksumSHA1": "glCzRHKY3rNJua7zOZrYeHkJ3DE=",
+			"path": "github.com/aws/aws-sdk-go-v2/service",
```
Comment (kaiyan-sheng) [inline vendor/vendor.json]: Hmmm why are we importing the whole service directory here?

Comment (andrewvc) [reply vendor/vendor.json]: Good catch

Comment (andrewvc) [reply vendor/vendor.json]: Just removed it!

**Comment by andrewvc** [comment]
@kaiyan-sheng just removed that extra dir.

@ruflin I'm on the fence WRT manual testing. @kaiyan-sheng / @ph WDYT? Are the automated tests not sufficient here?

**Comment by ruflin** [comment]
@andrewvc I didn't mean to manually test before we can merge this PR, but before the next release. AFAIK we don't do direct integrations tests with the service yet, so I think we should do this.

File: vendor/vendor.json [hunk 2]
```diff
@@ -247,6 +253,12 @@
 			"version": "v2.0.0-preview.5",
 			"versionExact": "v2.0.0-preview.5"
 		},
+		{
+			"checksumSHA1": "YfL/pY39NR7bx8DcINuneXWFjg4=",
+			"path": "github.com/aws/aws-sdk-go-v2/internal",
```
Comment (kaiyan-sheng) [inline vendor/vendor.json]: Sorry Im seeing this is also a directory that includes several subdirectories: awstesting, awsutill, ini, s3err, sdk and shareddefaults. Do we need to import all? I see that we are already importing awsutil, ini, sdk already.

File: vendor/vendor.json [hunk 3]
```diff
@@ -271,6 +283,12 @@
 			"version": "v2.0.0-preview.5",
 			"versionExact": "v2.0.0-preview.5"
 		},
+		{
+			"checksumSHA1": "oBbcBm1YWxduO3qRDLI5ZOfRVvw=",
+			"path": "github.com/aws/aws-sdk-go-v2/private",
+			"revision": "db223f6d5a456d1c9dfaeb7ef865cc63e03ec2ce",
```
Comment (kaiyan-sheng) [inline vendor/vendor.json]: same here, sorry!

File: vendor/vendor.json (hunk 2 above)
Comment (andrewvc) [reply vendor/vendor.json]: This  seems to be out of date with the latest push.

File: vendor/vendor.json (hunk 3 above)
Comment (andrewvc) [reply vendor/vendor.json]: I think this is now used by many of the other API packages with the upgrade

**Comment by andrewvc** [comment]
@kaiyan-sheng @ph updated the PR using a new pinned dep to the 0.9.0 (preview 9 in AWS lib parlance) version.

File: vendor/vendor.json [hunk 4]
```diff
@@ -420,23 +461,46 @@
...
+		{
+			"checksumSHA1": "kCyrtNPVHzoajcxIYke+pHqWkR0=",
+			"path": "github.com/aws/aws-sdk-go-v2/service/elasticloadbalancingv2/elasticloadbalancingv2iface",
+			"revision": "db223f6d5a456d1c9dfaeb7ef865cc63e03ec2ce",
+			"revisionTime": "2019-06-11T23:54:58Z",
+			"tree": true,
+			"version": "v2.0.0-preview.9"
+		},
+		{
+			"path": "github.com/aws/aws-sdk-go-v2/service/elasticloadbalancingv2iface",
```
Comment (kaiyan-sheng) [inline vendor/vendor.json]: Hmm I can't find this path `github.com/aws/aws-sdk-go-v2/service/elasticloadbalancingv2iface` for elasticloadbalancingv2iface in https://github.com/aws/aws-sdk-go-v2/tree/master/service

File: vendor/vendor.json (hunk 3 above)
Comment (kaiyan-sheng) [reply vendor/vendor.json]: So if https://github.com/elastic/beats/pull/12548/files#diff-bd290170e2912d3e8694db1a151066e5R298 and https://github.com/elastic/beats/pull/12548/files#diff-bd290170e2912d3e8694db1a151066e5R266 are deleted, this PR will fail?

**Comment by kaiyan-sheng** [comment]
Sorry one super basic question: where do you see the `v2.0.0-preview.9` in https://github.com/aws/aws-sdk-go-v2? I can only see up to `v2.0.0-preview.5` hmmm
Screen Shot 2019-06-17 at 4 55 58 PM

**Comment by andrewvc** [comment]
Closing in favor of https://github.com/elastic/beats/pull/12614

**Pull request closed without merging**